app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -0,0 +1,50 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    # Generic JWT authenticator that receive JWT vendor configuration and uses to validate that the authentication
+    # request is valid, and return conjur authn token accordingly
+    Authenticate = CommandClass.new(
+      dependencies: {
+        token_factory: TokenFactory.new
+      },
+      inputs: %i[jwt_configuration authenticator_input]
+    ) do
+
+      def call
+        validate_and_decode_token
+        get_conjur_id
+        validate_restrictions
+        new_token
+      end
+
+      private
+
+      def validate_and_decode_token
+        unless @jwt_configuration.validate_and_decode_token
```

Comment:
how come `validate_and_decode_token` does not need the token from the `authenticator_input`?

Reply:
Added  @authenticator_input.credentials that containts buddy of the request.
Will be changed out of scope of this pr accoridng to UX decisions